WooCommerce/Classes/ViewRelated/Orders/OrderStatusListViewController.swift

```diff
@@ -0,0 +1,249 @@
+import UIKit
+import Yosemite
+
+final class OrderStatusListViewController: UIViewController {
+    /// Main TableView.
+    ///
+    @IBOutlet private var tableView: UITableView!
+
+    /// ResultsController: Surrounds us. Binds the galaxy together. And also, keeps the UITableView <> (Stored) OrderStatuses in sync.
+    ///
+    private lazy var statusResultsController: ResultsController<StorageOrderStatus> = {
+        let storageManager = AppDelegate.shared.storageManager
+        let predicate = NSPredicate(format: "siteID == %lld", StoresManager.shared.sessionManager.defaultStoreID ?? Int.min)
+        let descriptor = NSSortDescriptor(key: "slug", ascending: true)
+
+        return ResultsController<StorageOrderStatus>(storageManager: storageManager, matching: predicate, sortedBy: [descriptor])
+    }()
+
+    /// The status selected
+    ///
+    private var selectedStatus: OrderStatus? {
+        didSet {
+            activateApplyButton()
+        }
+    }
+
+    /// Pull To Refresh Support.
+    ///
+    private lazy var refreshControl: UIRefreshControl = {
+        let refreshControl = UIRefreshControl()
+        refreshControl.addTarget(self, action: #selector(pullToRefresh(sender:)), for: .valueChanged)
+        return refreshControl
+    }()
+
+    /// Order to be provided with a new status
+    ///
+    private let order: Order
+
+    init(order: Order) {
+        self.order = order
+        super.init(nibName: type(of: self).nibName, bundle: nil)
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        registerTableViewCells()
+        configureNavigationBar()
+        configureTableView()
+
+        configureResultsController()
+    }
+
+    /// Setup: Results Controller
+    ///
+    private func configureResultsController() {
+        statusResultsController.startForwardingEvents(to: tableView)
+        try? statusResultsController.performFetch()
+    }
+
+    /// Registers all of the available TableViewCells
+    ///
+    private func registerTableViewCells() {
+        let cells = [StatusListTableViewCell.self]
+
+        for cell in cells {
+            tableView.register(cell.loadNib(), forCellReuseIdentifier: cell.reuseIdentifier)
+        }
+    }
+
+    /// Setup: TableView
+    ///
+    func configureTableView() {
+        view.backgroundColor = StyleManager.tableViewBackgroundColor
+        tableView.backgroundColor = StyleManager.tableViewBackgroundColor
+        tableView.refreshControl = refreshControl
+
```

Comment:
More Woo styling: don't forget to change the tableview in the .xib file to be a `grouped` table style.